File: applications/event-account-jdbc-sink/src/main/java/com/vmware/account/jdbc/sink/domain/AccountEntity.java
package com.vmware.account.jdbc.sink.domain;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "evt_accounts")
@Data
@NoArgsConstructor
public class AccountEntity {
    @Id
    private String id;

    private String name;
    private String accountType;
    private String status;
    private String notes;

    @OneToOne(cascade = CascadeType.ALL )
    private LocationEntity location ;

}

File: applications/account-jdbc-amqp-sink/src/main/kotlin/com/vmware/tanzu/data/services/rabbitmq/streaming/account/domain/AccountEntity.kt
package com.vmware.tanzu.data.services.rabbitmq.streaming.account.domain

import javax.persistence.*

@Entity
@Table(name = "evt_accounts")
data class AccountEntity(@Id var id : String = "",
                         var name: String = "",
                         var accountType : String = "",
                         var status : String  = "",
                         var notes : String = "",
                         @OneToOne(cascade = [CascadeType.ALL])
                   var location: LocationEntity = LocationEntity()
)
